storages/jsonl-storage/src/lib.rs

```diff
@@ -0,0 +1,419 @@
+        prelude::{parse, translate, Key},
+        result::{Error, Result},
+        store::{DataRow, RowIter, Store},
+        {chrono::NaiveDateTime, store::StoreMut},
+    },
+    serde_json::{Map, Value as JsonValue},
+    std::{
+        collections::HashMap,
+        fmt,
+        fs::{self, remove_file, File, OpenOptions},
+        io::{self, prelude::*, BufRead},
+        path::{Path, PathBuf},
+    },
+    utils::HashMapExt,
+};
+
+#[derive(Debug)]
+pub struct JsonlStorage {
+    pub path: PathBuf,
+}
+
+trait ResultExt<T, E: ToString> {
+    fn map_storage_err(self) -> Result<T, Error>;
+}
+
+impl<T, E: ToString> ResultExt<T, E> for std::result::Result<T, E> {
+    fn map_storage_err(self) -> Result<T, Error> {
+        self.map_err(|e| e.to_string()).map_err(Error::StorageMsg)
+    }
+}
+
+trait OptionExt<T> {
+    fn map_storage_err(self, payload: String) -> Result<T, Error>;
+}
+
+impl<T> OptionExt<T> for std::option::Option<T> {
+    fn map_storage_err(self, payload: String) -> Result<T, Error> {
+        self.ok_or_else(|| payload.to_string())
+            .map_err(Error::StorageMsg)
+    }
+}
+
+enum JsonlStorageError {
+    FileNotFound,
+    TableDoesNotExist,
+    ColumnDoesNotExist,
+    WrongSchemaFile(String),
+}
+
+impl fmt::Display for JsonlStorageError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let payload = match self {
+            JsonlStorageError::FileNotFound => "file not found".to_owned(),
+            JsonlStorageError::TableDoesNotExist => "table does not exist".to_owned(),
+            JsonlStorageError::ColumnDoesNotExist => "column does not exist".to_owned(),
+            JsonlStorageError::WrongSchemaFile(schema_path) => {
+                format!("schema file is wrong: {schema_path}")
+            }
+        };
+
+        write!(f, "{}", payload)
+    }
+}
+
+impl JsonlStorage {
+    pub fn new(path: &str) -> Result<Self> {
+        fs::create_dir_all(path).map_storage_err()?;
+        let path = PathBuf::from(path);
+
+        Ok(Self { path })
+    }
+
+    fn fetch_schema(&self, table_name: &str) -> Result<Option<Schema>> {
+        if !self.data_path(table_name).exists() {
+            return Ok(None);
+        };
+
+        let schema_path = self.schema_path(table_name);
+        let column_defs = match schema_path.exists() {
+            true => {
+                let mut file = File::open(&schema_path).map_storage_err()?;
+                let mut ddl = String::new();
+                file.read_to_string(&mut ddl).map_storage_err()?;
+
+                let parsed = parse(ddl)?.into_iter().next().map_storage_err(
+                    JsonlStorageError::WrongSchemaFile(
+                        schema_path
+                            .to_str()
+                            .map_storage_err(JsonlStorageError::FileNotFound.to_string())?
+                            .to_string(),
+                    )
+                    .to_string(),
+                )?;
```

Comment:
https://coveralls.io/builds/56635692/source?filename=storages%2Fjsonl-storage%2Fsrc%2Flib.rs#L101
I don't get why it was not covered. Kinda bug or actually unreachable..?